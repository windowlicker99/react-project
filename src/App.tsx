import React, { FC } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { AppRouter } from '@routing/AppRouter';

export const App: FC = () => (
  <Router>
    <AppRouter />
  </Router>
);
