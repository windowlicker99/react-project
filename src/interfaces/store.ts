import { ETableTypes, IColumn, ICurrentElement, IElementsTabsColumnsList, IUser } from '@interfaces/interfaces';

export interface IAuth {
  user: IUser;
}
export interface IPagination {
  nextPage: () => void;
  prevPage: () => void;
  pickPage: (indexPage: number) => void;
  disablePrev: boolean;
  disableNext: boolean;
  currentPage: number;
  pageAmount: number;
  currentTab: string;
  totalElement: number;
  totalElementOnPage: number;
  showedElements: number;
}

export interface IVehicleStore {
  currentVehicle: string;
}

export interface IHeaderStore {
  subtitles: string[];
}

export interface IChangeTypeAction {
  type: ETableTypes;
}

export interface IUpdateDisplayedColumnsAction {
  displayColumnsList: IDisplayColumnsList;
}

export interface ICustomizeColumnState {
  type: ETableTypes;
  columns: IElementsTabsColumnsList;
}

export interface IStoreState {
  auth: IAuth;
  pagination: IPagination;
  vehicles: IVehicleStore;
  customizeColumns: ICustomizeColumnState;
  header: IHeaderStore;
  element: IElementStore;
}

export interface IDisplayColumnsList {
  [key: string]: boolean;
}

export interface ICustomizeColumnAction {
  type: ETableTypes;
  displayColumnsList: IDisplayColumnsList;
}

export interface ICustomizeColumnsState {
  [key: string]: IColumn[];
}

export interface IElementStore {
  currentElement: ICurrentElement;
}
