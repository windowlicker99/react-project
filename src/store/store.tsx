import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import { authReducer } from '@store/slices/authSlice';
import { columnsReducer } from '@store/slices/columnsSlice';
import { paginationReducer } from '@store/slices/paginationSlice';
import { vehiclesReducer } from '@store/slices/vehicleSlice';
import { headerReducer } from '@store/slices/headerSlice';
import { elementReducer } from '@store/slices/elementSlice';

const reducers = combineReducers({
  auth: authReducer,
  vehicles: vehiclesReducer,
  customizeColumns: columnsReducer,
  pagination: paginationReducer,
  header: headerReducer,
  element: elementReducer,
});

const persistConfig = {
  key: 'root',
  storage,
  whiteList: ['auth', 'customizeColumns'],
};

const persistedReducer = persistReducer(persistConfig, reducers);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware({
      serializableCheck: false,
    }),
  ],
});

export const persistor = persistStore(store);
