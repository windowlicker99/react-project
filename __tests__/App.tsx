import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { store } from '../src/store/store';
import { App } from '../src/App';

describe('App component tests', () => {
  // it('Should render', () => {
  //   const { getByText } = render(<App />);
  //   const title = getByText(/App page/i);
  //   expect(title).toBeTruthy();
  // });

  it('Should match shapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
