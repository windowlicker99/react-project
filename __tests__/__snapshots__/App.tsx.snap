// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests Should match shapshot 1`] = `
<DocumentFragment>
  <div
    class="page-container is-open"
  >
    <div
      class="login-image"
    >
      <a
        class="logo"
        href="/"
      >
        <span
          class="icon-logo"
        />
      </a>
    </div>
    <div
      class="login-content"
    >
      <form
        action="#"
        class="login-form"
      >
        <span
          class="login-title"
        >
          Login to start
        </span>
        <div
          class="form-wrap"
        >
          <div
            class="form-group"
          >
            <label
              class="form-label"
            >
              Username
            </label>
            <input
              autocomplete="on"
              class="form-control"
              form="[object Object]"
              id="name"
              maxlength="15"
              name="name"
              placeholder="Enter Username"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="form-wrap"
        >
          <div
            class="form-group"
          >
            <label
              class="form-label"
            >
              Password
            </label>
            <input
              autocomplete="on"
              class="form-control"
              form="[object Object]"
              id="password"
              maxlength="50"
              name="password"
              placeholder="Enter Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          class="btn btn-lg-primary mb-30"
          disabled=""
          type="submit"
        >
          <span
            class="btn-icon "
          />
          <span
            class="btn-text"
          >
            Log In
          </span>
        </button>
        <div
          class="d-flex justify-content-end"
        >
          <a
            class="link"
            href="#"
          >
            <span
              class="nill mr-7"
            />
            Forgot Password?
          </a>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
